### ISSUE TITLE
Support NGINX's `auth_request` directly in Unit

### ISSUE THREAD

**Issue by lolgab** [nginx/unit#704]
Can Unit support something like [NGINX's auth_request](https://nginx.org/en/docs/http/ngx_http_auth_request_module.html) without the need of running Unit behind a NGINX reverse proxy?
Thank you!

**Comment by tippexs** [comment]
Hi @lolgab. At the moment, there is no nativ implementation of `auth_request` but I a  little python helper app would do the trick. Is that an option? If so we could use this thread to discuss the options you have.

**Comment by lolgab** [comment]
@tippexs Yes, that may also work. I will try this route and let's see how it goes :)

**Comment by tippexs** [comment]
Thanks for sharing your idea. The most simple idea would be a implemenation with Cookies and the use of our routing matches. Another - more powerful - implemenation could be made by the use of njs. We have a beta version available using NJS on Unit. We could try to make it work with njs on unit. Whats the best use case? APIs or a real website in which we can add cookies for example?

**Comment by lolgab** [comment]
Sorry, I forgot to mention my specific use case! I want to delegate authentication to [vouch-proxy](https://github.com/vouch/vouch-proxy), a microservice that handles authentication using the NGINX `auth_request` protocol.
It is nice since you can centralize the authentication to multiple applications running in the same Unit.
I haven't tried yet, but since it is written in Go, it might run in Unit as well.

**Comment by BradleyKirton** [comment]
Hi @tippexs, could you provide an example of what the unit configuration file would look like for your suggestion of using a python application for authentication?
